crates/chain/src/keychain/txout_index.rs

```diff
@@ -126,14 +210,31 @@ impl<K> KeychainTxOutIndex<K> {
     pub fn new(lookahead: u32) -> Self {
         Self {
             inner: SpkTxOutIndex::default(),
-            keychains: BTreeMap::new(),
+            descriptors: BTreeMap::new(),
+            descriptor_ids_to_keychain: BTreeMap::new(),
+            keychains_to_descriptor_ids: BTreeMap::new(),
             last_revealed: BTreeMap::new(),
             lookahead,
         }
     }
 }
 
 impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
+    /// Returns the corresponding (descriptor_id, descriptor) given a keychain
+    fn descriptor_of_keychain(
+        &self,
+        keychain: &K,
+    ) -> (&DescriptorId, &Descriptor<DescriptorPublicKey>) {
+        let descriptor_id = self
+            .keychains_to_descriptor_ids
+            .get(keychain)
+            .expect("keychain must exist");
+        let descriptor = self
+            .descriptors
+            .get(descriptor_id)
+            .expect("descriptor must exist");
+        (descriptor_id, descriptor)
+    }
```

Comment:
We need to be careful about calling this. Ideally, we avoid panics. If that is not possible, we need to document panics.

Reply:
Good catch. I changed descriptor_of_keychain to return an Option, and I explicitly expected where needed.

I am thinking of also panicking in `revealed_spks_of_keychain` and `unused_spks_of_keychain` - previously, they would just return empty iterators if you passed in a non-existent keychain, but panicking seems more appropriate. What do you think?

Reply:
I prefer returning empty iterators instead of panicing. I'm thinking if there was no descriptor under a keychain, nothing should be returned when we query about the keychain.

Reply:
Are there any real use cases for passing in a non-existent keychain? I can't think of any, and if that's the case, then passing in a non-existent keychain is a mistake, and mistakes should be reported explicitly to the user. I agree that panicking is a bit dramatic, it does make the API a bit cleaner compared than returning Result or Option (which I would be ok with, but it seems to me that we don't want to do that? I'm not sure why.)

In any case, we should be consistent across methods of the keychaintxout index, as currently the methods that return iterators return an empty one if you pass in a non-existent descriptor, while the others panic.

Reply:
I think it will make sense to return an `Option` if there is no descriptor under the keychain,.

Reply:
I realized it's better to leave it as is in this PR, and open an issue and discuss possible solutions there.